### PR TITLE
Data generator reorganisation

### PR DESCRIPTION
**Pull request by shounakmk219**
## Description
This PR decouples the data writer part from the data generator so that we can easily add new writers by keeping data generator untouched.

## Labels
`refactor`

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/AvroRecordAppender.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumWriter;
+
+
+public class AvroRecordAppender implements Closeable {
+
+  private final DataFileWriter<GenericData.Record> _recordWriter;
+  private final org.apache.avro.Schema _avroSchema;
+
+  public AvroRecordAppender(File file, org.apache.avro.Schema avroSchema)
```
Comment (xiangfu0) [inline pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/AvroRecordAppender.java:35]: Is this class reusable? Or shall we merge it into AvroWriter?

File: pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/FileWriter.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        writer.append(headers).append('\n');
+        for (int j = 0; j < numPerFiles; j++) {
+          String appendString = generateRow(_spec.getGenerator());
+          writer.append(appendString).append('\n');
+        }
+      }
+    }
+  }
+
+  protected String generateRow(DataGenerator generator) {
```
Comment (xiangfu0) [inline pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/FileWriter.java:50]: Make this method/class abstract?

**Review by xiangfu0 — APPROVED** [review]
overall lgtm, can you please add more comments to the interfaces you added?

File: pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/FileWriter.java (hunk 2 above)
Comment (shounakmk219) [reply pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/FileWriter.java:50]: Agree, will make the change.

File: pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/AvroRecordAppender.java (hunk 1 above)
Comment (shounakmk219) [reply pinot-controller/src/main/java/org/apache/pinot/controller/recommender/data/writer/AvroRecordAppender.java:35]: Atleast right now only AvroWriter is using it so makes sense to merge👍

**Comment by shounakmk219** [comment]
Thanks for the review @xiangfu0 !
I have addressed your comments along with a bugfix for removing hardcoded file extension and also added a new `cleanup()` method to `Writer` interface.

**Pull request merged**